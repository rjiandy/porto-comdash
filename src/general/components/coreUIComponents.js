// @flow
import Badge from './core-ui/Badge';
import Button from './core-ui/Button';
import Dropdown from './core-ui/Dropdown';
import ErrorComponent from './core-ui/ErrorComponent';
import FixedRatioView from './core-ui/FixedRatioView';
import FlexImage from './core-ui/FlexImage';
import MaterialIcon, {Icon} from './core-ui/Icon';
import Image from './core-ui/Image';
import Link from './core-ui/Link';
import LoadingIndicator from './core-ui/LoadingIndicator';
import ScrollView from './core-ui/ScrollView';
import TabBar from './core-ui/TabBar';
import Text from './core-ui/Text';
import TextInput from './core-ui/TextInput';
import ToggleSwitch from './core-ui/Switch';
import View from './core-ui/View';
import DatePicker from './core-ui/DatePicker';
import Tag from './core-ui/Tag';
import Form from './core-ui/Form';
import {Tabs, Tab} from './core-ui/Tabs';

export {
  // Please always keep the list below sorted using fn + f5
  Badge,
  Button,
  DatePicker,
  Dropdown,
  ErrorComponent,
  FixedRatioView,
  FlexImage,
  Form,
  Icon,
  Image,
  Link,
  LoadingIndicator,
  MaterialIcon,
  ScrollView,
  TabBar,
  Tab,
  Tabs,
  Tag,
  Text,
  TextInput,
  ToggleSwitch,
  View,
};
