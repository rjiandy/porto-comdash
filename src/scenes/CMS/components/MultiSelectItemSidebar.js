// @flow
import React, {Component} from 'react';
import autobind from 'class-autobind';
import {Checkbox} from 'material-ui';
import {StyleSheet, Touchable} from 'react-primitives';
import {
  View,
  Button,
  TextInput,
  Text,
  MaterialIcon,
  Icon,
  ScrollView,
} from '../../../general/components/coreUIComponents';

import {
  TEXT_COLOR,
  THEME_COLOR,
  LIGHT_GREY,
  ALTERNATIVE_GREY,
  MEDIUM_GREY,
  MAIN_BLUE,
} from '../../../general/constants/colors';
import {
  SMALL_FONT_SIZE,
  DEFAULT_FONT_FAMILY,
} from '../../../general/constants/text';

type State = {
  selectedItems: Set<string>;
  filterValue: string;
  isAssignedFilterActive: boolean;
};

type ItemID = string;

type Item = {
  id: ItemID;
  name: string;
};
type Props = {
  icon?: string;
  searchPlaceholder: string;
  noItemFound: string;
  itemList: Array<Item>;
  selectedItems: Array<ItemID>;
  onConfirm: (selectedItems: Array<ItemID>) => void;
  toggleSidebar: (command: 'open' | 'close', data?: Array<ItemID>) => void;
  onAddData?: () => void;
  addDataButtonLabel?: string;
};

export type MultiSelectItemSidebarProps = Props;

export default class MultiSelectItemSidebar extends Component {
  state: State;
  props: Props;

  constructor() {
    super(...arguments);
    autobind(this);

    this.state = {
      selectedItems: this._getSelectedItems(this.props.selectedItems),
      filterValue: '',
      isAssignedFilterActive: false,
    };
  }

  componentWillReceiveProps(newProps: Props) {
    let oldProps = this.props;
    if (newProps.selectedItems.length !== oldProps.selectedItems.length) {
      this.setState({
        selectedItems: this._getSelectedItems(newProps.selectedItems),
      });
    }
  }
  _isTickAll(itemList: Array<Item>) {
    let {selectedItems} = this.state;
    for (let {id} of itemList) {
      if (!selectedItems.has(id)) {
        return false;
      }
    }
    if (itemList.length === 0) {
      return false;
    }
    return true;
  }
  _tickAll(itemList: Array<Item>) {
    let {selectedItems} = this.state;
    let newSelectedItems = new Set(selectedItems);
    let isTickAll = this._isTickAll(itemList);
    let isAssignedFilterActive = false;
    if (!isTickAll) {
      for (let {id} of itemList) {
        if (!selectedItems.has(id)) {
          newSelectedItems.add(id);
        }
      }
      isAssignedFilterActive = true;
    } else {
      for (let {id} of itemList) {
        newSelectedItems.delete(id);
      }
    }
    this.setState({
      selectedItems: newSelectedItems,
      isAssignedFilterActive,
    });
  }
  render() {
    let {selectedItems, filterValue, isAssignedFilterActive} = this.state;
    let {
      searchPlaceholder,
      itemList,
      toggleSidebar,
      onAddData,
      addDataButtonLabel,
      noItemFound,
      icon,
    } = this.props;

    let filteredItemList = itemList.filter(
      (item) =>
        item.name.toLowerCase().includes(filterValue.toLowerCase()) &&
        (isAssignedFilterActive
          ? selectedItems.has(item.id)
          : !selectedItems.has(item.id)),
    );

    return (
      <View style={styles.container}>
        <View style={{paddingHorizontal: 11, marginBottom: 10}}>
          <TextInput
            fullWidth
            hintText={searchPlaceholder}
            value={filterValue}
            onTextChange={(newFilterValue) =>
              this.setState({filterValue: newFilterValue})}
            icon="search"
            iconStyle={{color: LIGHT_GREY}}
          />
        </View>
        <View style={styles.itemsHeader}>
          <View style={{width: 40}}>
            <Checkbox
              uncheckedIcon={
                <MaterialIcon
                  name="check-box-outline-blank"
                  color={LIGHT_GREY}
                />
              }
              checkedIcon={
                <MaterialIcon name="check-box" style={{color: THEME_COLOR}} />
              }
              checked={this._isTickAll(filteredItemList)}
              onCheck={() => this._tickAll(filteredItemList)}
              style={styles.checkboxIcon}
            />
          </View>
          <View style={[styles.flexRow, styles.switchContainer]}>
            <Touchable
              onPress={() => this.setState({isAssignedFilterActive: true})}
            >
              <View style={styles.switchItemContainer}>
                <Text
                  style={{
                    fontSize: 10,
                    fontWeight: 'bold',
                    color: isAssignedFilterActive ? MAIN_BLUE : MEDIUM_GREY,
                  }}
                >
                  ASSIGNED
                </Text>
              </View>
            </Touchable>
            <Touchable
              onPress={() => this.setState({isAssignedFilterActive: false})}
            >
              <View style={styles.switchItemContainer}>
                <Text
                  style={{
                    fontSize: 10,
                    fontWeight: 'bold',
                    color: isAssignedFilterActive ? MEDIUM_GREY : MAIN_BLUE,
                  }}
                >
                  UNASSIGNED
                </Text>
              </View>
            </Touchable>
          </View>
        </View>
        <View style={styles.groupList}>
          {filteredItemList.length === 0 ? (
            <View style={styles.switchItemContainer}>
              <Text style={{textAlign: 'center', color: LIGHT_GREY}}>
                {noItemFound}
              </Text>
            </View>
          ) : (
            <ScrollView>
              {filteredItemList.map((item, index) => {
                return (
                  <View style={styles.rowItem} key={index}>
                    <Checkbox
                      uncheckedIcon={
                        <MaterialIcon
                          name="check-box-outline-blank"
                          color={LIGHT_GREY}
                        />
                      }
                      checkedIcon={
                        <MaterialIcon
                          name="check-box"
                          style={{color: THEME_COLOR}}
                        />
                      }
                      checked={selectedItems.has(item.id)}
                      onCheck={() => this._selectItem(item.id)}
                      style={StyleSheet.flatten([
                        styles.checkboxIcon,
                        {width: 40},
                      ])}
                    />
                    <View style={styles.checklistContent}>
                      <View
                        style={{
                          width: 20,
                          height: 20,
                          marginHorizontal: 8,
                        }}
                      >
                        <Icon name={icon || 'user'} />
                      </View>
                      <Text>{item.name}</Text>
                    </View>
                  </View>
                );
              })}
            </ScrollView>
          )}
        </View>
        <View style={{marginTop: 10}}>
          {onAddData && addDataButtonLabel ? (
            <Button
              secondary
              label={addDataButtonLabel}
              style={styles.button}
              onPress={onAddData}
            />
          ) : null}

          <View style={styles.bottomButtons}>
            <Button
              secondary
              label="Cancel"
              style={styles.button}
              onPress={() =>
                toggleSidebar('close', [...this.state.selectedItems.values()])}
            />
            <Button
              primary
              label="Confirm"
              style={StyleSheet.flatten([styles.button, {marginLeft: 15}])}
              onPress={() => this._onConfirm()}
            />
          </View>
        </View>
      </View>
    );
  }

  _getSelectedItems(selectedItems: Array<ItemID>) {
    return new Set(
      this.props.itemList
        .filter((item) => selectedItems.includes(item.id))
        .map((item) => item.id),
    );
  }

  _selectItem(item: string) {
    let {selectedItems} = this.state;
    if (selectedItems.has(item)) {
      selectedItems.delete(item);
    } else {
      selectedItems.add(item);
    }
    this.setState({selectedItems});
  }

  _onConfirm() {
    let {onConfirm, toggleSidebar} = this.props;
    let {selectedItems} = this.state;
    onConfirm(Array.from(selectedItems.values()));
    toggleSidebar('close', []);
  }
}

const styles = {
  container: {
    flex: 1,
  },
  flexRow: {
    flexDirection: 'row',
  },
  switchItemContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  switchContainer: {
    height: '100%',
    width: '80%',
    paddingVertical: 8,
  },
  itemsHeader: {
    height: 46,
    width: '90%',
    alignItems: 'center',
    marginHorizontal: 10,
    marginBottom: 6,
    borderBottomWidth: 1,
    borderBottomColor: ALTERNATIVE_GREY,
    flexDirection: 'row',
  },
  rowItem: {
    width: '90%',
    marginHorizontal: 10,
    marginBottom: 16,
    flexDirection: 'row',
    alignItems: 'center',
  },
  groupList: {
    flex: 1,
    paddingTop: 10,
  },
  bottomButtons: {
    flexDirection: 'row',
    marginTop: 10,
  },
  button: {
    flex: 1,
  },
  checkbox: {
    paddingLeft: 12,
    marginBottom: 10,
    marginHorizontal: 10,
    alignSelf: 'stretch',
    height: 42,
  },
  checkboxIcon: {
    color: THEME_COLOR,
    width: 20,
  },
  checklistContent: {
    flexDirection: 'row',
    borderWidth: 1,
    borderRadius: 2,
    borderColor: LIGHT_GREY,
    width: '80%',
    overflow: 'scroll', // TODO: Improve this styling later
    alignItems: 'center',
    paddingVertical: 10,
    paddingRight: 4,
  },
  checkboxLabel: {
    fontFamily: DEFAULT_FONT_FAMILY,
    fontSize: SMALL_FONT_SIZE,
    color: TEXT_COLOR,
  },
};
