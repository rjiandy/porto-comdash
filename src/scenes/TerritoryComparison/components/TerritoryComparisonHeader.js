// @flow
import React from 'react';
import {StyleSheet} from 'react-primitives';
import {
  View,
  TabBar,
} from '../../../general/components/coreUIComponents';

type NavigationProps = {
  selectedViewMode: string;
  tabList: Array<string>;
  onChangeViewMode: (newTab: string) => void;
};

function TerritoryComparisonHeader(props: NavigationProps) {
  let {
    selectedViewMode,
    tabList,
    onChangeViewMode,
  } = props;

  let formatedTabList = tabList.map((tabName) => {
    return {
      tabName,
    };
  });

  return (
    <View style={styles.filterContent}>
      <View style={styles.filterContentView}>
        <TabBar
          selectedTabName={selectedViewMode}
          tabList={formatedTabList}
          onChangeTab={onChangeViewMode}
        />
      </View>
    </View>
  );
}

let styles = StyleSheet.create({
  filterContent: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    justifyContent: 'space-between',
  },
  filterContentView: {
    paddingBottom: 20,
  },
});

export default TerritoryComparisonHeader;
