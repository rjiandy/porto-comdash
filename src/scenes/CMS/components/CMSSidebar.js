// @flow

import React from 'react';
import {StyleSheet} from 'react-primitives';
import {View, Text, Button} from '../../../general/components/coreUIComponents';
// import {DataLabel} from '../../../general/components/UIComponents';
import {THEME_COLOR, GREY} from '../../../general/constants/colors';

type Props = {
  isOpen: boolean;
  title: string;
  children?: ReactNode;
  onSave?: () => void; // don't need this if we are using form
  onCancelPress?: () => void;
  saveButtonProps?: any;
  cancelButtonProps?: any;
  style?: StyleSheetTypes;
  isSaveDisabled?: boolean;
};

export default function AddGroupSidebar(props: Props) {
  let {
    isOpen,
    title,
    children,
    onSave,
    saveButtonProps,
    cancelButtonProps,
    onCancelPress,
    isSaveDisabled,
    style,
  } = props;
  return isOpen ? (
    <View style={[styles.container, style]}>
      <Text customStyle="title" style={styles.sideBarTitle}>
        {title}
      </Text>
      <View style={styles.separatorBar} />
      {children}
    </View>
  ) : (
    <View style={[styles.container, style]}>
      <Button
        primary
        label="Save"
        fullWidth
        onPress={onSave}
        style={{marginBottom: 10}}
        {...saveButtonProps}
        backgroundColor={isSaveDisabled ? GREY : THEME_COLOR}
        disabled={isSaveDisabled}
      />
      <Button
        style={{marginBottom: 10}}
        secondary
        label="Cancel"
        fullWidth
        onPress={onCancelPress}
        {...cancelButtonProps}
      />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: 'white',
    maxWidth: 260, // weird behaviour because it keeps growing
    minWidth: 260,
    flex: 1,
    padding: 20,
    paddingBottom: 12,
    borderLeftWidth: 1,
    borderColor: '#D0DAE3',
  },
  sideBarTitle: {
    alignSelf: 'center',
    marginBottom: 10,
  },
  separatorBar: {
    height: 3,
    backgroundColor: THEME_COLOR,
    borderRadius: 2,
    marginBottom: 10,
    marginHorizontal: 10,
  },
});
