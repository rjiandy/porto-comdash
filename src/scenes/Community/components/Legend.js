// @flow
import React from 'react';
import {View, Text} from '../../../general/components/coreUIComponents';
import {StyleSheet} from 'react-primitives';

type LegendDatum = {
  color: string;
  text: string;
};

type LegendProps = {
  legends: Array<LegendDatum>;
};

function Legend(props: LegendProps) {
  let {legends} = props;
  let legendsNode = legends.map(({color, text}, id) => {
    return (
      <View key={id} style={styles.textAndBoxLegendWrapper}>
        <View
          style={StyleSheet.flatten([
            styles.legendColorBox,
            {backgroundColor: color},
          ])}
        />
        <Text>
          {text}
        </Text>
      </View>
    );
  });
  return (
    <View style={styles.legendContainer}>
      {legendsNode}
    </View>
  );
}

let styles = StyleSheet.create({
  legendContainer: {
    flexDirection: 'row',
    padding: 10,
  },
  textAndBoxLegendWrapper: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    marginRight: 20,
  },
  legendColorBox: {
    height: 10,
    width: 10,
    marginRight: 5,
  },
});

export default Legend;
