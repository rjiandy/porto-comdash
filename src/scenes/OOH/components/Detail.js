// @flow

import React from 'react';
import {StyleSheet} from 'react-primitives';
import {View, Text} from '../../../general/components/coreUIComponents';
import {PlaceholderView} from '../../../general/components/UIComponents';
import PieChart from '../../../general/components/charts/PieChart';
import BarChart from '../../../general/components/charts/BarRechart';

import {THEME_COLOR, MEDIUM_GREY} from '../../../general/constants/colors';

import type {OOHDetailData} from '../type/OOH-type';

type Props = {
  data: OOHDetailData;
  selectedTerritory: string;
  selectedType: string;
};

export default function Detail(props: Props) {
  let {data, selectedTerritory, selectedType} = props;

  let filteredData = data.filter((datum) => datum.oohType === selectedType);

  let siteData = filteredData.filter((datum) => datum.itemType === 'SITE');

  let visualData = filteredData.filter((datum) => datum.itemType === 'VISUAL');

  let siteDataContent;
  if (siteData.length > 0) {
    siteDataContent = (
      <View style={[styles.chartItem, styles.borderContainer]}>
        <Text customStyle="title">By Site</Text>
        <PieChart
          data={siteData}
          xAxis="label"
          yAxis="measure"
          showTooltip
          showLegend
          showLabelValue
        />
      </View>
    );
  } else {
    siteDataContent = (
      <PlaceholderView
        text={`No site data available for territory ${selectedTerritory}, OOH type ${selectedType}`}
      />
    );
  }

  let visualDataContent;
  if (visualData.length > 0) {
    visualDataContent = (
      <View style={[styles.chartItem, styles.borderContainer]}>
        <Text customStyle="title">By Visual</Text>
        <BarChart
          data={visualData}
          xAxis="label"
          yAxis="measure"
          horizontal
          showTooltip
          showLabelValue
          showChartGrid
          extraProps={{
            bar: {
              fill: THEME_COLOR,
            },
          }}
        />
      </View>
    );
  } else {
    visualDataContent = (
      <PlaceholderView
        text={`No visual data available for territory ${selectedTerritory}, OOH type ${selectedType}`}
      />
    );
  }
  return (
    <View style={styles.root}>
      <View style={styles.flex}>
        {!selectedType ? (
          <PlaceholderView text="Please select OOH type first" />
        ) : (
          <View style={styles.chartContainer}>
            {siteDataContent}
            {visualDataContent}
          </View>
        )}
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  root: {
    flex: 1,
    flexDirection: 'row',
    paddingTop: 10,
  },
  flex: {
    flex: 1,
  },
  chartContainer: {
    flex: 1,
    flexDirection: 'row',
  },
  noDataAvailable: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    paddingHorizontal: 20,
  },
  noDataAvailableText: {
    textAlign: 'center',
  },
  chartItem: {
    flex: 1,
    marginHorizontal: 5,
    alignItems: 'center',
    justifyContent: 'center',
  },
  borderContainer: {
    borderRadius: 5,
    borderWidth: 1,
    borderColor: MEDIUM_GREY,
  },
});
