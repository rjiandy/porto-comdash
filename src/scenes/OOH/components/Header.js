// @flow

import React from 'react';
import {StyleSheet} from 'react-primitives';

import {View, Text} from '../../../general/components/coreUIComponents';
import {PlaceholderView} from '../../../general/components/UIComponents';
import commaSeparator from '../../../general/helpers/commaSeparator';
import {THEME_COLOR, GREY} from '../../../general/constants/colors';
import {HEADER_FONT_SIZE} from '../../../general/constants/text';

import type {OOH} from '../type/OOH-type';

type Props = {
  data: ?OOH;
  selectedTerritory: string;
};

export default function Header(props: Props) {
  let {data, selectedTerritory} = props;
  if (!selectedTerritory) {
    return (
      <View style={{height: 80}}>
        <PlaceholderView text="Please select territory first" />
      </View>
    );
  } else if (!data) {
    return (
      <View style={{height: 80}}>
        <PlaceholderView
          text={`No data available for territory ${selectedTerritory}`}
        />
      </View>
    );
  }

  return (
    <View style={styles.header}>
      <View style={styles.headerItem}>
        <Text
          style={[styles.headerData, styles.headerDataNumber]}
          customStyle="header"
        >
          {commaSeparator(data.billboard || 0)}
        </Text>
        <Text style={styles.headerData} customStyle="title">
          Billboard
        </Text>
      </View>
      <View style={styles.headerItem}>
        <Text
          style={[styles.headerData, styles.headerDataNumber]}
          customStyle="header"
        >
          {commaSeparator(data.baliho || 0)}
        </Text>
        <Text style={styles.headerData} customStyle="title">
          Baliho
        </Text>
      </View>
      <View style={styles.headerItem}>
        <Text
          style={[styles.headerData, styles.headerDataNumber]}
          customStyle="header"
        >
          {commaSeparator(data.minibillboard || 0)}
        </Text>
        <Text style={styles.headerData} customStyle="title">
          Mini Billboard
        </Text>
      </View>
      <View style={styles.headerItem}>
        <Text
          style={[styles.headerData, styles.headerDataNumber]}
          customStyle="header"
        >
          {commaSeparator(data.streetsignage || 0)}
        </Text>
        <Text style={styles.headerData} customStyle="title">
          Street Sign
        </Text>
      </View>
      <View style={styles.headerItem}>
        <Text
          style={[styles.headerData, styles.headerDataNumber]}
          customStyle="header"
        >
          {commaSeparator(data.others || 0)}
        </Text>
        <Text style={styles.headerData} customStyle="title">
          Other OOH
        </Text>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  headerItem: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  header: {
    flexDirection: 'row',
    borderWidth: 5,
    borderColor: THEME_COLOR,
    borderStyle: 'solid',
    paddingVertical: 10,
    paddingHorizontal: 30,
  },
  headerData: {
    color: GREY,
    fontWeight: 'bold',
  },
  headerDataNumber: {
    fontSize: HEADER_FONT_SIZE + 10,
  },
});
