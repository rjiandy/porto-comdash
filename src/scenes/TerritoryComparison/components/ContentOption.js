// @flow

import React from 'react';
import {StyleSheet} from 'react-primitives';

import {View, TabBar} from '../../../general/components/coreUIComponents';

type ContentOptionProps = {
  selectedZoneMode: string;
  zoneModeList: Array<string>;
  onZoneChange: (newZone: string) => void;
  selectedVolumeMode: string;
  volumeModeList: Array<string>;
  onVolumeModeChange: (newVolumeMode: string) => void;
};

function ContentOption(props: ContentOptionProps) {
  let {
    selectedZoneMode,
    zoneModeList,
    onZoneChange,
    selectedVolumeMode,
    volumeModeList,
    onVolumeModeChange,
  } = props;
  let convertFromArrayToTabListFormat = (arr) =>
    arr.map((name) => ({tabName: name}));
  let zoneList = convertFromArrayToTabListFormat(zoneModeList);
  let volumeList = convertFromArrayToTabListFormat(volumeModeList);
  return (
    <View style={styles.toggleContentContainer}>
      <View style={{marginBottom: 10}}>
        <TabBar
          selectedTabName={selectedZoneMode}
          tabList={zoneList}
          onChangeTab={onZoneChange}
        />
      </View>
      <View>
        <TabBar
          selectedTabName={selectedVolumeMode}
          tabList={volumeList}
          onChangeTab={onVolumeModeChange}
        />
      </View>
    </View>
  );
}

let styles = StyleSheet.create({
  toggleContentContainer: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    height: 100,
  },
});

export default ContentOption;
