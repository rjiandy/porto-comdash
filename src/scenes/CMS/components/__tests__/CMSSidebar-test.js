// @flow

import React from 'react';
import renderer from 'react-test-renderer';

import {View} from '../../../../general/components/coreUIComponents';
import CMSSidebar from '../CMSSidebar';

jest.mock('../../../../general/components/coreUIComponents', () => ({
  Text: 'Text',
  View: 'View',
}));

it('should render a CMS Sidebar with Correct Children', () => {
  let AddGroupSidebar = View;
  expect(
    renderer
      .create(
        <CMSSidebar isOpen={true} title="Group List">
          <AddGroupSidebar props={{}} />
        </CMSSidebar>
      )
      .toJSON()
  ).toMatchSnapshot();
});
