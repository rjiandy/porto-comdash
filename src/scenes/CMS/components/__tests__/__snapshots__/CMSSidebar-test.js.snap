// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a CMS Sidebar with Correct Children 1`] = `
<View
  style={
    Array [
      67,
      undefined,
    ]
  }
>
  <Text
    customStyle="title"
    style={68}
  >
    Group List
  </Text>
  <View
    style={69}
  />
  <View
    props={Object {}}
  />
</View>
`;
